src/pages/index.js

```diff
@@ -31,64 +121,114 @@ const enableValidation = (config) => {
 
 enableValidation(config);
 
-// POPUP ELEMENTS
-
-const imagePreviewPopup = new PopupWithImage("#modal-preview-image");
-
-const addCardPopup = new PopupWithForm("#modal-add-card", handleCardFormSubmit);
-
-const editProfilePopup = new PopupWithForm(
-  "#modal-profile-edit",
-  handleProfileFormSubmit
-);
-
-const userInfo = new UserInfo(
-  document.querySelector(".profile__name"),
-  document.querySelector(".profile__subtitle")
-);
-
-// RENDER CARD FUNCTIONS
+/*
+  ┌─────────────────────────────────────────────────────────────────────────┐
+  │ RENDER CARD FUNCTIONS                                                   │
+  └─────────────────────────────────────────────────────────────────────────┘
+ */
 
 function handleCardClick({ name, link }) {
   imagePreviewPopup.open(name, link);
 }
 
 function renderCard(cardData) {
-  const cardElement = new Card(cardData, "#card-template", handleCardClick);
+  const cardElement = new Card(
+    cardData,
+    "#card-template",
+    handleCardClick,
+    function handleDeleteClick() {
+      cardDeletePopup.setSubmitAction(() => {
+        cardDeletePopup.setLoading(true);
+        api
+          .removeCard(cardData._id)
+          .then((res) => {
+            cardElement.removeCardElement(res._id);
+            cardDeletePopup.close();
+          })
+          .catch((err) => {
+            console.error(err);
+          })
+          .finally(() => {
+            cardDeletePopup.setLoading(false, "Yes");
+          });
+      });
+      cardDeletePopup.open();
+    },
+    function handleLikeButtonClick() {
+      api
+        .changeLikeStatus(cardData._id, cardElement.isLiked)
+        .then((res) => {
+          console.log(res);
+          cardElement.updateLikes(res.isLiked);
+        })
+        .catch((err) => {
+          console.error(err);
+        });
+    }
+  );
   return cardElement.getView();
 }
 
-// RENDER INITAL CARDS
+/*
+  ┌─────────────────────────────────────────────────────────────────────────┐
+  │ EVENT HANDLERS                                                          │
+  └─────────────────────────────────────────────────────────────────────────┘
+ */
+
+function handleProfileFormSubmit({ name, description }) {
+  editProfilePopup.setLoading(true);
+  api
+    .updateUserInfo(name, description)
+    .then(() => {
+      userInfo.setUserInfo(name, description);
+      editProfilePopup.close();
+    })
+    .catch((err) => {
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors